From: George McCollister <george.mccollister@gmail.com>
Date: Fri, 23 Mar 2012 13:13:41 -0500
Subject: [PATCH] Fill out Installed-Size

Fill out Installed-Size so opkg can check to make sure there is enough
space to install the package before attempting to do so.

Signed-off-by: George McCollister <george.mccollister@gmail.com>
---
 opkg.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/opkg.py b/opkg.py
index 153b62c7e7fa..0ee9853b2ba6 100644
--- a/opkg.py
+++ b/opkg.py
@@ -169,9 +169,13 @@ class Package(object):
             if tarfile.is_tarfile(fn):
                 tar = tarfile.open(fn, "r", f)
                 tarStream = tar.extractfile("./control.tar.gz")
+                dataTarStream = tar.extractfile("./data.tar.gz")
             else:
                 ar = arfile.ArFile(f, fn)
                 tarStream = ar.open("control.tar.gz")
+                dataTarStream = ar.open("data.tar.gz")
+            tarf = tarfile.open("data.tar.gz", "r", dataTarStream)
+            self.installed_size = sum([f.size for f in tarf.getmembers()])
             tarf = tarfile.open("control.tar.gz", "r", tarStream)
             try:
                 control = tarf.extractfile("control")
@@ -482,7 +486,7 @@ class Package(object):
         if self.maintainer: out = out + "Maintainer: %s\n" % (self.maintainer)
         if self.md5: out = out + "MD5Sum: %s\n" % (self.md5)
         if self.size: out = out + "Size: %d\n" % int(self.size)
-        if self.installed_size: out = out + "InstalledSize: %d\n" % int(self.installed_size)
+        if self.installed_size: out = out + "Installed-Size: %d\n" % int(self.installed_size)
         if self.filename: out = out + "Filename: %s\n" % (self.filename)
         if self.source: out = out + "Source: %s\n" % (self.source)
         if self.description: out = out + "Description: %s\n" % (self.description)
